controllers/btpoperator_controller_test.go

```diff
@@ -500,6 +516,313 @@ var _ = Describe("BTP Operator controller", Ordered, func() {
+				Expect(ok).To(BeTrue())
+				Expect(!bytes.Equal(currentCaCert, beforeCaCert))
+
+				currentWebhookSecret := getSecret(WebhookSecret)
+				currentWebhookCert, ok := currentWebhookSecret.Data[reconciler.buildKeyNameWithExtension(WebhookSecretDataPrefix, CertificatePostfix)]
+				Expect(ok).To(BeTrue())
+				beforeWebhookCert, ok := beforeWebhookSecret.Data[reconciler.buildKeyNameWithExtension(WebhookSecretDataPrefix, CertificatePostfix)]
+				Expect(ok).To(BeTrue())
+				Expect(!bytes.Equal(currentWebhookCert, beforeWebhookCert))
+				Expect(!bytes.Equal(currentWebhookCert, newWebhookCertificate))
+
+				ensureCorrectState()
+			})
+		})
+
+		When("webhook certificate expires", func() {
+			fakeTime := 30.0
+			fakeExpiration := 10.0
+			BeforeAll(func() {
+				timeOpts := &certificationsTimeOpts{
+					CaCertificateExpiration: CaCertificateExpiration,
+					WebhookCertExpiration:   time.Second * time.Duration(fakeTime),
+					ExpirationBoundary:      time.Second * time.Duration(fakeExpiration),
+				}
+				certBeforeEach(timeOpts)
+			})
+			AfterAll(func() {
+				certAfterEach()
+			})
+
+			It("CA certificate is not changed, webhook certificate is regenerated", func() {
+				caSecretBeforeExpiration := getSecret(CaSecret)
+				webhookSecretBeforeExpiration := getSecret(WebhookSecret)
+				Expect(checkHowManySecondsToExpiration(WebhookSecret) <= fakeTime).To(BeTrue())
+				restoreOriginalCertificateTiems()
+				ensureReconcilationQueueIsEmpty()
+				_, err := reconciler.Reconcile(ctx, controllerruntime.Request{NamespacedName: apimachienerytypes.NamespacedName{
+					Namespace: cr.Namespace,
+					Name:      cr.Name,
+				}})
+				Expect(err).To(BeNil())
+				ensureReconcilationQueueIsEmpty()
+				caSecretAfterExpiration := getSecret(CaSecret)
+				webhookSecretAfterExpiration := getSecret(WebhookSecret)
+				Expect(reflect.DeepEqual(caSecretBeforeExpiration.Data, caSecretAfterExpiration.Data)).To(BeTrue())
+				Expect(reflect.DeepEqual(webhookSecretBeforeExpiration.Data, webhookSecretAfterExpiration.Data)).To(BeFalse())
+				Expect(checkHowManySecondsToExpiration(WebhookSecret) >= fakeTime).To(BeTrue())
+
+				ensureCorrectState()
+			})
+		})
+
+		When("CA certificate expires", func() {
+			fakeSeconds := 30.0
+			fakeExpiration := 10.0
+			BeforeAll(func() {
+				timeOpts := &certificationsTimeOpts{
+					CaCertificateExpiration: time.Second * time.Duration(fakeSeconds),
+					WebhookCertExpiration:   orgWebhookCertExpiration,
+					ExpirationBoundary:      time.Second * time.Duration(fakeExpiration),
+				}
+				certBeforeEach(timeOpts)
+			})
+			AfterAll(func() {
+				certAfterEach()
+			})
+
+			It("fully regenerate of CA certificate and webhook certificate", func() {
+				caSecretBeforeExpiration := getSecret(CaSecret)
+				webhookSecretBeforeExpiration := getSecret(WebhookSecret)
+				Expect(checkHowManySecondsToExpiration(CaSecret) <= fakeSeconds).To(BeTrue())
+				time.Sleep(time.Second * 10)
```

Comment:
you can poll `time.After` to check whether cert expired instead of using `time.Sleep`